pkg/proxy/proxier.go

```diff
@@ -548,12 +555,33 @@ func (proxier *Proxier) openPortal(service types.NamespacedName, info *serviceIn
 	if err != nil {
 		return err
 	}
+
+	publicIpCount := 0
 	for _, publicIP := range info.publicIP {
-		err = proxier.openOnePortal(net.ParseIP(publicIP), info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
+		ip := net.ParseIP(publicIP)
+		if ip == nil {
+			glog.V(2).Info("Skipping non-address public endpoint: %s", publicIP)
+			continue
+		}
+		publicIpCount++
+		err = proxier.openOnePortal(ip, info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
 		if err != nil {
 			return err
 		}
 	}
+
+	// We have an external load balancer, but no specific IPs.
```

Comment:
(I will address the other comments later, but this one seems bigger-picture...)

You are correct: as implemented here, there is a global limit of one service per port, though this uniqueness constraint applies only to externally load balanced services.

These unique ports don't have to be the same as the public-facing ports though: ELB can listen on port 80, and forward to port 8080.  So one not-terrible (though more code needed) option would be to reserve a range of ports that kube-proxy would listen on, and then assign external-load-balanced services from that range.  This would allow two services to run with (public) port 80, with two external load balancers, with ELB1 mapping 80 -> 8081, and ELB2 mapping 80 -> 8082.  This port mapping cannot be different per-instance, so the assignment would have to be done at the cluster-level, rather than per-kube-proxy.

There may also be a trick/hack to figuring out which load balancer sourced the traffic (ELB does support the proxy protocol, which is tantalizingly close to giving us what we need).  I don't know of any such trick, but will investigate.

A hangout seems higher bandwidth; happy to do one next week.